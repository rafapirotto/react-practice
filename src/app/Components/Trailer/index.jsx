import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  title: PropTypes.string,
  url: PropTypes.string,
};

const defaultProps = {
  title: '',
  url: '',
};

const renderTitle = (title) => (
  <div>
    <h1 className="ml-24 text-3xl font-semibold">{title}</h1>
  </div>
);

const renderTrailer = (url) => (
  <iframe
    className="flex justify-center mt-24 ml-24"
    width="560"
    height="315"
    src={url}
    title="YouTube video player"
    frameBorder="0"
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen
  />
);

const pascalCase = (s) => s.replaceAll('_', ' ');

const Trailer = ({ title, url }) => (
  <main>
    {renderTitle(`${pascalCase(title)}'s trailer`)}
    {renderTrailer(url)}
  </main>
);

Trailer.propTypes = propTypes;
Trailer.defaultProps = defaultProps;

export default Trailer;
