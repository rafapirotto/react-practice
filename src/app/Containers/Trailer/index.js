import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import axios from 'axios';

import { Trailer } from '../../Components';

const TrailerContainer = ({ match }) => {
  const {
    params: { movieId },
  } = match;

  const { content } = useSelector((state) => state.popularMovies);
  const [movie, setMovie] = useState(null);

  const fetchMovie = async (movieId) => {
    const { data: movie } = await axios.get(
      `${process.env.REACT_APP_BASE_URL}/${process.env.REACT_APP_MOVIES_URL}/${movieId}`,
    );

    setMovie(movie);
  };

  useEffect(() => {
    if (!content) {
      fetchMovie(movieId);
    } else {
      const storeMovie = content.find(({ id }) => id === parseInt(movieId));

      setMovie(storeMovie);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <main>
      <Trailer title={movie?.title} url={movie?.trailer} />
    </main>
  );
};

export default TrailerContainer;
