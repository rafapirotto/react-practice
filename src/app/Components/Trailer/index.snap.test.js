import React from 'react';
import renderer from 'react-test-renderer';

import Trailer from '.';

test('It matches the snapshot', () => {
  const component = renderer.create(<Trailer title="title" url="some-url" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
