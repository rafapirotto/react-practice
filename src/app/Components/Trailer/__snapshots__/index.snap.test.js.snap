// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches the snapshot 1`] = `
Array [
  <div>
    <h1
      className="ml-24 text-3xl font-semibold"
    >
      title's trailer
    </h1>
  </div>,
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    className="flex justify-center mt-24 ml-24"
    frameBorder="0"
    height="315"
    src="some-url"
    title="title"
    width="560"
  />,
]
`;
